### DISCUSSION TITLE
Update from 13.1.0.4 to 14.3.0 stuck in safe mode

### DISCUSSION THREAD

**Discussion by Leseratte10** [arendst/Tasmota#22333] [Q&A]
I have four Shelly devices that I tried to OTA update from 13.1.0.4 to the current 14.3.0 using the URL http://ota.tasmota.com/tasmota32/release/tasmota32solo1.bin

Three of them worked just fine. The fourth one seems to have run into an issue during the update and is now in safeboot mode (13.1.0.4(safeboot)-2_0_14). 

Restarting doesn't get the device out of safeboot mode. When I try the update again, either using the OTA URL or by uploading the file, I can see in the console that it starts the update, and after a couple seconds it restarts with the following crash message in the log: 

14:40:15.384 MQT: tele/tasmota_2E98D4/INFO3 = {"Info3":{"RestartReason":{"Exception":29,"Reason":"StoreProhibited","EPC":"40082ae8","EXCVADDR":"00000000","CallChain":["40082ae5","4008c149","4008f7b5","4010218f","4008516d","4000bfed","4008c611","40164ac7","40122e66","40112008","40113c52","400e390d","400e3988","400e3a59","400e2ac2","400def1a","400def41","400f6b25","400eed00","400f95b7","400e21ed"]},"BootCount":60}}

I tried "Restart 3", no change. I tried "Upgrade 2" which I found [here](https://github.com/arendst/Tasmota/discussions/21113) but that says "1 or >13.1.0.4 to upgrade". Any idea why "Upgrade 2" fails?

If I go to "Information", it says

- Partition safeboot*: 832 KB (used 91%)
- Partition app0: 2880 KB
- Partition fs: 320 KB

Is there any chance to get this to work again without serial flashing? Unfortunately this device (Shelly) has an annoyingly small pin pitch (1.25mm?) instead of the standard 2.54 so it was difficult enough to do the initial flash months ago ...

**Comment by Leseratte10** [comment]
Whatever it was, after dozens of attempts and multiple reboots I was finally able to flash the new firmware from safeboot. 

Would still be interesting to know A) why that happened, B) why I was unable to update and C) if there's a way to prevent this in the future.

**Reply by nikito7** [reply]
Maybe safeboot it's too old.

Do upgrade 2 first.

Upgrade 2 only work from main partition. 

In some cases safeboot will be flashed to main partition

**Comment by rp2019433** [comment]
I noticed the same behavior when updating some ESP32-C3 devices. I got stuck in safemode and tried several times to update using OTA. It didn't work and I had to flash Tasmota through serial again.
Maybe there is something happening with the update process in new version.
